src/factors/BearingRange2D.jl

```diff
@@ -10,6 +10,16 @@ Bearing and Range constraint from a Pose2 to Point2 variable.
 mutable struct Pose2Point2BearingRange{B <: IIF.SamplableBelief, R <: IIF.SamplableBelief} <: IIF.AbstractRelativeMinimize
     bearing::B
     range::R
+    # buffer for parametric
+    Z::Union{MvNormal, Nothing}
+end
+
+Pose2Point2BearingRange(bearing, range) = Pose2Point2BearingRange(bearing, range, nothing)
+
+function Pose2Point2BearingRange(bearing::Normal, range::Normal) 
+  μ = mean.([bearing, range])
+  σ = var.([bearing, range])
+  Pose2Point2BearingRange(bearing, range, MvNormal(μ, σ))
 end
```

Comment:
@dehann, can something like this potentially cause problems for mm

Reply:
The unique reuse memory per factor will likely happen more and more (to avoid allocations).  Perhaps just use internal name like `_reuseZ` rather than just `Z` which I find bit too confusing (also wary of duplication).  Getting rid of duplicate factor residual functions is most important right now.  We can find our grove for a `._reuse` convention after IIF 467 is closed.

Reply:
```julia
function Pose2Point2BearingRange(bearing::Normal, range::Normal) 
```

this is pushing the limit a bit, but at least just drop the duplicate factor functions at this point.

Reply:
> can something like this potentially cause problems for mm

If the tests pass it should be fine, or we will discover which tests to add.

Reply:
Actually @Affie , the `::Union{}` inside the type is no good, please remove that to a single type only for type-stability.

Reply:
Thanks for having a look, the new this and masters test is passing.
I'll sleep on it, perhaps there is a better way of doing it.

Reply:
Looking at it again I'd say just use `._reuseGaussian::MvNormal`.  Don't worry about super optimizing all this just yet.  The most important thing now is to consolidate residual functions, once things are reduced to tests and a single logic definition code it becomes a lot easier to optimize / refactor.

Reply:
> Actually @Affie , the `::Union{}` inside the struct is no good, please remove that to a single type only for type-stability.

In general, I would rather have a `nothing` returned and than a valid `MvNormal`. An error should be thrown rather than the code looks like its working with some default MvNormal. That being said, I think this way overcomplicated it and I'm thinking it would be more general to overload `getParametricMeasuremet`.

Reply:
i also prefer overloading `getParametricMeasurement`